### ISSUE TITLE
Add Support for Parent Device Roles

### ISSUE THREAD

**Issue by djmelik** [netbox-community/netbox#12209]
### NetBox version

v3.4.7

### Feature type

Data model extension

### Proposed functionality

I would like for Netbox to be able to support parent-child relationships in the context of Device Roles. It would be nice to be able to nest device role types for greater granularity and organization.

### Use case

Currently, we have many different router and switch device role types. For instance; we have Border Router, Peering Router, Access Router, Core Router, Spine Router, Leaf Router, etc. It would be nice to be able to create a parent "Router" Device Role parent type, and set all granular router device roles as children. In other words, nest Access Router, Border Router, Peering Router, etc under Router.

This would significantly help with both filtering and just general organization. 

### Database changes

Unfortunately, I am not a developer; I do not know how to meaningfully answer this question. 

### External dependencies

N/A

**Comment by jeremystretch** [comment]
IMO this makes sense, but first we need to figure out what we're doing with MPTT, which is what's currently in place for all self-nesting models. I'm going to mark this as blocked by #6587 / #11421.

**Comment by jeremystretch** [comment]
Blocked by #12552

**Comment by DanSheps** [comment]
Duplicate of #6290
